libavcodec/x86/vvc_mc.asm

```diff
@@ -0,0 +1,1546 @@
+    vpackusdw        m9, m9, m11
+
+    vpunpcklwd      m10,  m8, m9
+    vpunpckhwd      m11,  m8, m9
+    vpunpckldq       m8, m10, m11
+    vpunpckhdq       m9, m10, m11
+
+    vextracti64x4  ym10, m8, 1
+    vextracti64x4  ym11, m9, 1
+
+    vpunpcklqdq    ym12, ym8, ym10
+    vpunpckhqdq    ym13, ym8, ym10
+    vpunpcklqdq    ym14, ym9, ym11
+    vpunpckhqdq    ym15, ym9, ym11
+
+    movu          [dstq + MAX_PB_SIZE * 0], xm12
+    movu          [dstq + MAX_PB_SIZE * 1], xm13
+    movu          [dstq + MAX_PB_SIZE * 2], xm14
+    movu          [dstq + MAX_PB_SIZE * 3], xm15
+    vextracti32x4 [dstq + MAX_PB_SIZE * 4], ym12, 1
+    vextracti32x4 [dstq + MAX_PB_SIZE * 5], ym13, 1
+    vextracti32x4 [dstq + MAX_PB_SIZE * 6], ym14, 1
+    vextracti32x4 [dstq + MAX_PB_SIZE * 7], ym15, 1
+
+    LOOP_END 8
+
+    RET
+
+.hv4:
+    LOAD_SHUFFLE_16 pw_vvc_iter_shuffle_index_quarter, 0, 1, 2, 3
+
+.loop_h4:
+    mov               xq, widthq
+.loop_v4:
+    mov             srcq, _srcq
+    lea           r3srcq, [srcstrideq * 3]
+    sub             srcq, r3srcq
+    vpxor           ym11, ym11, ym11
+    movu             xm8, [srcq                 ]
+    movu             xm9, [srcq + srcstrideq * 1]
+    movu            xm10, [srcq + srcstrideq * 2]
+    movu            xm11, [srcq + r3srcq        ]
+
+    movu            xm12, [srcq + srcstrideq * 4]
+
+    lea             srcq, [srcq + r3srcq        ]
+    movu            xm13, [srcq + srcstrideq * 2]
+
+    lea             srcq, [srcq + r3srcq        ]
+    movu            xm14, [srcq                 ]
+    movu            xm15, [srcq + srcstrideq * 1]
+
+    movu            xm16, [srcq + srcstrideq * 2]
+    movu            xm17, [srcq + r3srcq        ]
+    movu            xm18, [srcq + srcstrideq * 4]
+    vinserti64x2     ym8,  ym8, xm16, 1
+    vinserti64x2     ym9,  ym9, xm17, 1
+    vinserti64x2    ym10, ym10, xm18, 1
+
+    vpunpcklwd       ym8,  ym8,  ym9
+    vpunpcklwd      ym10, ym10, ym11
+    vpunpcklwd      ym12, ym12, ym13
+    vpunpcklwd      ym14, ym14, ym15
+
+    vpunpckldq       ym9,  ym8, ym10
+    vpunpckhdq      ym11,  ym8, ym10
+    vpunpckldq      ym10, ym12, ym14
+    vpunpckhdq      ym12, ym12, ym14
+
+    vpunpcklqdq      ym8,  ym9, ym10
+    vpunpckhqdq      ym9,  ym9, ym10
+    vpunpcklqdq     ym10, ym11, ym12
+    vpunpckhqdq     ym11, ym11, ym12
+
+    vinserti64x4      m8, m8, ym10, 1
+    vinserti64x4      m9, m9, ym11, 1
+
+    H_COMPUTE_H4_16    8, 9
+    vpmovdw          ym8,   m8
+    vextracti64x2    xm9,  ym8, 1
+    vpunpcklwd      xm10,  xm8, xm9
+    vpunpckhwd      xm11,  xm8, xm9
+    vpunpckldq       xm8, xm10, xm11
+    vpunpckhdq       xm9, xm10, xm11
+
+    psrldq          xm10, xm8, 8
+    psrldq          xm11, xm9, 8
+
+    movq [dstq + MAX_PB_SIZE * 0], xm8
+    movq [dstq + MAX_PB_SIZE * 1], xm10
+    movq [dstq + MAX_PB_SIZE * 2], xm9
+    movq [dstq + MAX_PB_SIZE * 3], xm11
+
+    LOOP_END 4
+
+    RET
+%endmacro
+
+%macro H_COMPUTE_16_AVX2 6
```

Comment:
better provide some comments for the macro header. and macro body. 
see https://code.videolan.org/videolan/dav1d/-/blob/master/src/x86/mc16_avx2.asm#L1356